src/Adapter/Tax/TaxComputer.php

```diff
@@ -43,17 +45,25 @@ class TaxComputer
     protected const DIVISION_PRECISION = Division::DEFAULT_PRECISION + 2;
 
     /**
-     * @var TaxRulesGroupRepository
```

Comment:
You removed the repository here to replace everything with legacy code which was the exact opposite idea of this service initially 😅 I don't mind if the previous `getTaxRulesGroupDetails` function is not adapted anymore and needs to be replaced

But at least the new code should follow the same spirit to replace legacy code, so you can probably add a new method or maybe modify the existing `getTaxRulesGroupDetails` to meet your needs

And as I already suggested in other comments for a refacto to avoid duplicating all the logic related with the first state, I think this class and method would actually be good candidates And as I said in another comment, the language has nothing to do here since we don't need localized values It's only here because it's used `getTaxRulesByGroupId` but most of the returned data has no interest here so instead you can implement a new request in the `TaxRulesGroupRepository` Either it returns all the states and you pick the first here, or it returns the first one automatically like `getCountryDefaultState` which you might need in the form type as well to adapt the wording

Reply:
Or alternatively maybe the `getTaxRulesGroupDetails` could return tax rate values for each states the country contains also, or it could have an optional stateId parameter to specify the targetted state